1-WebApp-OIDC/1-5-B2C/Startup.cs

```diff
@@ -56,15 +57,23 @@ public void ConfigureServices(IServiceCollection services)
             });
 
             // Configuration to sign-in users with Azure AD B2C
-            services.AddAuthentication(AzureADB2CDefaults.AuthenticationScheme)
-                .AddAzureADB2C(options => Configuration.Bind("AzureAdB2C", options));
+            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
+                .AddSignIn("AzureAdB2C", Configuration, options => Configuration.Bind("AzureAdB2C", options));
+
+            services.Configure<OpenIdConnectOptions>(OpenIdConnectDefaults.AuthenticationScheme, options => 
+            {
+                // The AddSignIn method sets NameClaimType to preferred_username but since B2C doesnt have this claim, we have to change it to name
+                options.TokenValidationParameters.NameClaimType = "name";
```

Comment:
The `AddSignIn` method sets `NameClaimType` to preferred_username but since B2C doesnt have this claim, we have to change it to something else after `AddSignIn` configuration.

Our B2C playground apps uses `name` for this claim.

Reply:
Will the developer need to make sure they also have this set up in the portal? or use a different value? 

---
In reply to: [371528695](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/272#discussion_r371528695) [](ancestors = 371528695)

Reply:
@jennyf19 they don't need to make sure, since `name` is present on B2C sisu by default. If they want to display the email for instance, they will have to set a different value (which will be `email` claims). 

Should I bring this configuration to Microsoft.Identity,Web instead? Then developers only need to add this line of code if they need something different than `name`?

Reply:
@tiago Brenck so, just to clarify, you would move this to MS.Identity.Web and it would be on as a default? I'm okay with that, but am not sure how many developers are using the default susi policy. I think most use custom policies...but i'll all for making their lives easier. @jmprieur what do you think?

---
In reply to: [372645393](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/272#discussion_r372645393) [](ancestors = 372645393)

Reply:
@jennyf19 I moved this part to Identity.Web, since we already set `preferred_username` for AAD scenarios. If developers don't have the claims `name` in their user flows, they still have the option to overwrite Identity.Web configuration and set `NameClaimType` to the claim that fits their needs.

I think the way it is now, makes their lives easier if they don't want to change it, but also gives a way to change this value if needed.